examples/pytorch/graphsage/link_pred.py

```diff
@@ -69,23 +75,26 @@ def forward(self, pair_graph, neg_pair_graph, blocks, x):
     def inference(self, g, device, batch_size, num_workers, buffer_device=None):
         # The difference between this inference function and the one in the official
         # example is that the intermediate results can also benefit from prefetching.
-        g.ndata['h'] = g.ndata['feat']
-        sampler = dgl.dataloading.MultiLayerFullNeighborSampler(1, prefetch_node_feats=['h'])
+        feat = g.ndata['feat']
+        sampler = dgl.dataloading.MultiLayerFullNeighborSampler(1, prefetch_node_feats=['feat'])
         dataloader = dgl.dataloading.NodeDataLoader(
                 g, torch.arange(g.num_nodes()).to(g.device), sampler, device=device,
                 batch_size=1000, shuffle=False, drop_last=False, num_workers=num_workers)
         if buffer_device is None:
             buffer_device = device
 
         for l, layer in enumerate(self.layers):
-            y = torch.zeros(g.num_nodes(), self.n_hidden, device=buffer_device)
+            y = torch.zeros(g.num_nodes(), self.n_hidden, device=buffer_device,
+                            pin_memory=args.pure_gpu)
+            feat = feat.to(device)
+
             for input_nodes, output_nodes, blocks in tqdm.tqdm(dataloader):
-                x = blocks[0].srcdata['h']
+                x = feat[input_nodes]
```

Comment:
@BarclayII I thought your previous implementation used prefetching for intermediate results. Is it critical? Does this change break that?

Reply:
I think it has performance impacts.  Usability-wise it should be OK.